File: src-tauri/src/models/chapter.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(tag = "type", rename_all = "lowercase")]
pub enum ChapterStatus {
    Unread,
    Ongoing { page: usize },
    Read,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Chapter {
    pub id: String,
    pub title: String,
    pub path: String,
    pub file_type: String,
    pub page_count: usize,
    pub status: ChapterStatus,
}

File: src/types/chapter.ts
export type ChapterStatus =
  | { type: "unread" }
  | { type: "ongoing"; page: number }
  | { type: "read" };

export interface Chapter {
  id: string;
  title: string;
  path: string;
  file_type: "dir" | "cbz";
  page_count: number;
  status: ChapterStatus;
}
